src/CLI/test.ts

```diff
@@ -27,19 +28,29 @@ describe("should compile tests project", () => {
 		}),
 	);
 	const program = createProjectProgram(data);
+	const compilerOptions = program.getCompilerOptions();
 	const pathTranslator = createPathTranslator(program, data);
 
 	// clean outDir between test runs
 	fs.removeSync(program.getCompilerOptions().outDir!);
 
 	it("should copy include files", () => copyInclude(data));
 
-	it("should copy non-compiled files", () =>
-		copyFiles(data, pathTranslator, new Set(getRootDirs(program.getCompilerOptions()))));
+	it("should copy non-compiled files", () => copyFiles(data, pathTranslator, new Set(getRootDirs(compilerOptions))));
+
+	const changedSourceFiles = getChangedSourceFiles(program);
+	const changedCompilableFiles = changedSourceFiles.filter(v => !v.isDeclarationFile);
+
+	it("should transform .d.ts files in rootDir and write to outDir", () => {
+		if (compilerOptions.declaration) {
+			const changedDeclarationFiles = changedSourceFiles.filter(v => v.isDeclarationFile);
+			transformAndWriteDeclarationFiles(program.getProgram(), pathTranslator, changedDeclarationFiles);
+		}
+	});
```

Comment:
We should actually test this; if possible also checking that it transforms things as expected, as opposed to just copying.

Reply:
I couldn't figure out how. You'd have to first compile something and then try to use it? It wouldn't matter if the .d.ts is in the same src directory, because you wouldn't use it's transformed form at all.